core/src/main/java/org/jobrunr/storage/sql/common/DatabaseCreator.java

```diff
@@ -213,7 +253,136 @@ private TablePrefixStatementUpdater getStatementUpdater(String tablePrefix, Conn
+            try (final Connection conn = getConnection(); final PreparedStatement pSt = conn.prepareStatement("select * from " + tablePrefixStatementUpdater.getFQTableName("jobrunr_migrations") + " where id = ?")) {
+                pSt.setString(1, TABLE_LOCKER_UUID);
+                ResultSet rs = pSt.executeQuery();
+                if (rs.next()) {
+                    // why: we want to confirm that the migrations are still running, see also this::startMigrationsTableLockUpdateTimer()
+                    Instant lastTableLockUpdate = parse(rs.getString("installedOn"));
+                    if (now().isAfter(lastTableLockUpdate.plus(20, ChronoUnit.SECONDS))) {
+                        throw new IllegalStateException("Database migrations have timed out.");
+                    }
+                    return true;
+                }
+                return false;
+            }
+        }
+
+        private void insertLock(Connection connection) throws SQLException {
+            try (final PreparedStatement pSt = connection.prepareStatement("insert into " + tablePrefixStatementUpdater.getFQTableName("jobrunr_migrations") + " values (?, ?, ?)")) {
+                pSt.setString(1, TABLE_LOCKER_UUID);
+                pSt.setString(2, TABLE_LOCKER_SCRIPT);
+                pSt.setString(3, now().truncatedTo(MICROS).toString());
+                int updateCount = pSt.executeUpdate();
+                if (updateCount == 0) throw new IllegalStateException("Another DatabaseCreator is performing the migrations table.");
+            }
+        }
+
+        // why: dropping and creating new indexes can take a good amount of time. Here we update the installedOn column so it can be awaited and monitored by other servers.
+        private void updateLock(Connection connection) throws SQLException {
+            try (final PreparedStatement pSt = connection.prepareStatement("update " + tablePrefixStatementUpdater.getFQTableName("jobrunr_migrations") + " set installedOn = ? where id = ? and script = ?")) {
+                pSt.setString(1, now().truncatedTo(MICROS).toString());
+                pSt.setString(2, TABLE_LOCKER_UUID);
+                pSt.setString(3, TABLE_LOCKER_SCRIPT);
+                int updateCount = pSt.executeUpdate();
+                if (updateCount == 0) throw shouldNotHappenException(new IllegalStateException("Another DatabaseCreator is performing the migrations table."));
+            }
+        }
+
+        private void removeLock(Connection conn) {
+            try (final PreparedStatement pSt = conn.prepareStatement("delete from " + tablePrefixStatementUpdater.getFQTableName("jobrunr_migrations") + " where id = ?")) {
+                pSt.setString(1, TABLE_LOCKER_UUID);
+                int updateCount = pSt.executeUpdate();
+                if (updateCount == 0) throw shouldNotHappenException(new IllegalStateException("Another DatabaseCreator had the migrations table lock."));
```

Comment:
```suggestion
                if (updateCount == 0) throw new IllegalStateException("The migrations table lock has already been removed.");
```

Reply:
Why: I think the exception will be caught again and re-thrown as a `JobRunrException` and the message is more accurate